<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>lizard_gwas</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="fullcontent">

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">lizard_gwas</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="genome-scan-for-positive-selection" class="level2">
<h2 class="anchored" data-anchor-id="genome-scan-for-positive-selection">Genome scan for positive selection</h2>
<p>We are attempting to detect positive selection in the fence lizard genomes from haplotype distributions. Initially, we planned to use saltiLassi as described by DeGiorgio and Szpiech 2022. Due to some complications with the calculation of sliding window size, we are exploring other options.</p>
<p>Most methods for detecting positive selection identify distortions in the spatial distribution of snps (i.e.&nbsp;certain snps may have an inflated frequency which can be seen in a site frequency spectrum). saltiLassi implements haplotypes, and therefore looks for distortions in the distribution of haplotypes in the genome. My understanding is that this allows us to measure or control for linkage disequilibrium. When certain snps are swept closer to fixation due to positive selection, nearby neutral alleles are more likely than distal alleles to be swept along. As a result, there is a tendency for long haplotypes to form that display lower diversity in the regions directly flanking positively selected snps. Lassi investigates these patterns by searching the genome at successively wider “windows” and comparing the haplotype distribution against a background distribution.</p>
<p>Classic measures for ancestral/derived haplotypes are EHH (?) and the integrated haplotype score (iHS). These statistics compute haplotype homozygosity in a local genome region.</p>
<p>Lassi calculates the T-statistic, a larger value of which suggests a harder sweep. saltiLassi takes this a step further by integrating the spatial distribution into the likelihood model. According to DeGiorgio and Szpiech 2022, larger windows have higher power for determining ancient sweeps whereas smaller windows could better identify recent sweeps.</p>
<section id="some-questions-i-have" class="level3">
<h3 class="anchored" data-anchor-id="some-questions-i-have">Some questions I have:</h3>
<ol type="1">
<li>Are we separating the populations in this analysis?</li>
<li>What outputs do we expect from this analysis?</li>
</ol>
</section>
</section>
<section id="stacks-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="stacks-pipeline">Stacks pipeline</h2>
<pre class="{unix}}"><code>#| echo: false

populations -V vcf -O dir [-M popmap] (filters) [--fstats] [-k [--sigma=150000] [--bootstrap [-N 100]]] (output formats)</code></pre>
<p><strong>--fstats</strong> — enable SNP and haplotype-based F statistics</p>
<p><strong>--fst_correction</strong> — specify a correction to be applied to Fst values: ‘p_value’. Default: off.</p>
<p><strong>-V,--in_vcf</strong> — path to an input VCF file.</p>
<p><strong>-M,--popmap</strong> — path to a population map. (Format is ’SAMPLE1 \t POP1 \n SAMPLE2</p>
<p><strong>-O,--out_path</strong> — path to a directory where to white the output files. (Required by -V; otherwise defaults to value of -P.)</p>
<section id="consider-data-filtering" class="level3">
<h3 class="anchored" data-anchor-id="consider-data-filtering">Consider data filtering</h3>
<ul>
<li><p><strong>-H,--filter-haplotype-wise</strong> — apply the above filters haplotype wise (unshared SNPs will be pruned to reduce haplotype-wise missing data).</p>
<h3 id="kernel-smoothing-algorithm" class="anchored">Kernel-smoothing algorithm:</h3>
<ul>
<li><p><strong>-k,--smooth</strong> — enable kernel-smoothed π, F<sub>IS</sub>, F<sub>ST</sub>, F<sub>ST</sub>’, and Φ<sub>ST</sub> calculations.</p></li>
<li></li>
</ul></li>
</ul>
<pre class="{unix}}"><code>#| echo: false
ls</code></pre>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>